### PR TITLE
feat: strategy variant test UI

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Strategy variant basic UI - will change but wanted to test it in sandbox for now. It's behind a feature flag.
Screenshot 2023-07-10 at 15 54 43

### CODE REVIEW

File: frontend/src/component/feature/StrategyTypes/FlexibleStrategy/FlexibleStrategy.tsx [hunk 1]
```diff
@@ -123,6 +126,35 @@ const FlexibleStrategy = ({
                     onChange={e => onUpdate('groupId')(e.target.value)}
                     data-testid={FLEXIBLE_STRATEGY_GROUP_ID}
                 />
+                <br />
```
Comment (Tymek) [inline frontend/src/component/feature/StrategyTypes/FlexibleStrategy/FlexibleStrategy.tsx:129]: FYI, line break added, not behind flag

File: src/lib/types/experimental.ts [hunk 2]
```diff
@@ -108,6 +109,7 @@ const flags: IFlags = {
         process.env.UNLEASH_EXPERIMENTAL_CUSTOM_ROOT_ROLES,
         false,
     ),
+    strategyVariant: parseEnvVarBoolean(process.env.STRATEGY_VARIANT, false),
```
Comment (Tymek) [inline src/lib/types/experimental.ts:112]: We usually add `UNLEASH_` prefix to environment variables.

File: frontend/src/component/feature/StrategyTypes/FlexibleStrategy/FlexibleStrategy.tsx [hunk 3]
```diff
@@ -123,6 +126,35 @@ const FlexibleStrategy = ({
...
+                <ConditionallyRender
+                    condition={Boolean(uiConfig?.flags?.strategyVariant)}
+                    show={
+                        <>
```
Comment (Tymek) [inline frontend/src/component/feature/StrategyTypes/FlexibleStrategy/FlexibleStrategy.tsx:133]: Good for now, probably should be a separate component later on.

**Review by Tymek — APPROVED** [review]

File: frontend/src/component/feature/StrategyTypes/FlexibleStrategy/FlexibleStrategy.tsx (hunk 3 above)
Comment (kwasniew) [reply frontend/src/component/feature/StrategyTypes/FlexibleStrategy/FlexibleStrategy.tsx:133]: This is temporary and design will change. Just wanted to demo this POC

**Pull request merged**